funnel/views/helpers.py

```diff
@@ -17,3 +23,40 @@ def localize_date(date, from_tz=utc, to_tz=utc):
             to_tz = pytz_timezone(to_tz)
         return from_tz.localize(date).astimezone(to_tz).replace(tzinfo=None)
     return date
+
+
+def location_geodata(location):
+    """ Same as from hasjob/views/helper.py
+        TODO: Extract into a common lib.
+    """
+    if 'HASCORE_SERVER' in app.config:
+        if isinstance(location, (list, tuple)):
+            url = urljoin(app.config['HASCORE_SERVER'], '/1/geo/get_by_names')
+        else:
+            url = urljoin(app.config['HASCORE_SERVER'], '/1/geo/get_by_name')
+        response = requests.get(url, params={'name': location}).json()
+        if response.get('status') == 'ok':
+            result = response.get('result', {})
+            if isinstance(result, (list, tuple)):
+                result = {r['geonameid']: r for r in result}
+            return result
+    return {}
+
+
+def parsed_location_geodata(data):
+    if 'HASCORE_SERVER' in app.config:
+        url = urljoin(app.config['HASCORE_SERVER'], '/1/geo/parse_locations')
+        response = requests.get(url, params={'q': data}).json()
+        return [field['geoname'] for field in response['result'] if 'geoname' in field.keys()][0]
+    return {}
```

Comment:
Actually, since the first returns a dict inside a list, you should disambiguate by returning the actual data that is interesting: the integer `geonameid` or `None`.